crates/turborepo-lib/src/rewrite_json.rs

```diff
@@ -0,0 +1,411 @@
+    let start;
+    let end;
+
+    match generate_type {
+        GenerateType::Object => {
+            separator = "";
+            start = range.start;
+            end = range.end;
+        }
+        GenerateType::Member => {
+            if property_count > 0 {
+                separator = ",";
+            } else {
+                separator = ""
+            }
+
+            start = range.start + 1;
+            end = range.start + 1;
+        }
+    }
+
+    // Generate the serialized JSON to insert into the document.
+    // We synthesize objects for missing path segments.
+    let missing_path_segments = path[closest_path.len()..].to_vec();
+    let computed_object = match generate_type {
+        GenerateType::Object => generate_object(missing_path_segments, json_value),
+        GenerateType::Member => generate_member(missing_path_segments, json_value, separator),
+    };
+
+    // Generate a new document!
+    let mut output: String = json_document_string.to_owned();
+    output.replace_range(start..end, &computed_object);
+
+    Ok(output)
+}
+
+/**
+ * get_root returns the document root, or information on the error
+ * encountered with the input json_document_string.
+ */
+fn get_root(json_document_string: &str) -> Result<jsonc_parser::ast::Value, RewriteError> {
+    let parse_result_result = parse_to_ast(
+        json_document_string,
+        &Default::default(),
+        &Default::default(),
+    );
+
+    match parse_result_result {
+        Ok(parse_result) => match parse_result.value {
+            Some(root) => Ok(root),
+            None => Err(RewriteError::NoRoot),
+        },
+        Err(parse_error) => Err(RewriteError::ParseError(parse_error)),
+    }
+}
+
+/**
+ * get_closest_node traverses the JSON document via recursive tail calls to
+ * find the last node that exists with in this JSON key path.
+ */
+fn get_closest_node<'a>(
+    current_node: &'a jsonc_parser::ast::Value<'a>,
+    target_path: &Vec<&'a str>,
+    current_path: Vec<&'a str>,
+) -> (Vec<&'a str>, &'a jsonc_parser::ast::Value<'a>) {
+    // No target_path? We've arrived.
+    if target_path.is_empty() {
+        return (current_path, current_node);
+    }
+
+    match current_node {
+        // Only objects can have key paths.
+        jsonc_parser::ast::Value::Object(obj) => {
+            // Grab the last property (member) which matches the current target_path
+            // element.
+            let object_property = obj.properties.iter().rev().find(|property| {
+                let current_property_name = property.name.as_str();
+                target_path[0] == current_property_name
+            });
+
+            // See if we found a matching key. If so, recurse.
+            match object_property {
+                Some(property) => {
+                    let next_current_node = &property.value;
+                    let next_property_name = property.name.as_str();
+                    let mut next_current_path = current_path.clone();
```

Comment:
I started by renaming the variable, and so that led me to a local maxima of `.clone()`. I really just wanted to alias the existing everything from that line forward.

Is `&mut *current_path` a reasonable pattern here?